### PR TITLE
Adds some chilling hallucinations

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Adds some new ice-based hallucinations
The basic freeze lasts about 6 seconds

https://user-images.githubusercontent.com/66052067/193428664-102250a5-202d-4d32-95a4-a9b9371e383d.mp4

The ice sound can also play at random as its own hallucination

The other hallucination adds-

https://user-images.githubusercontent.com/66052067/193428738-bcd319b5-7855-48d4-9481-9f7a3e25f573.mp4

You have a small window to run, but it never helps.

All [sounds](https://freesound.org/people/cognito%20perceptu/sounds/38693/) from [freesound](https://freesound.org/people/InspectorJ/sounds/420875/).org

## Why It's Good For The Game
[More hallucination variety](https://youtu.be/nC8zz95rm8s)

## Changelog

:cl: Wallem
add: A new ice-based hallucination has been added to the game
/:cl:

### CODE REVIEW

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/hallucination/ice_cube.dm [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	hallucinator.clear_alert(ALERT_TEMPERATURE, clear_override = TRUE)
+	REMOVE_TRAIT(hallucinator, TRAIT_IMMOBILIZED, HALLUCINATION_TRAIT)
+	if(ice_overlay)
+		hallucinator.client?.images -= ice_overlay
+		ice_overlay = null
+
+/datum/hallucination/ice/freezer
+	random_hallucination_weight = 0
+	ice_duration = 11 SECONDS
+	play_ice_sound = FALSE
```
[the remark this answers is not in the document]
Comment (Wallemations) [reply code/modules/hallucination/ice_cube.dm:45]: Initially I did that, but it ended up just making the entire thing not work at all. Making it a subtype was the fix

File: code/modules/hallucination/fake_sound.dm [hunk 2]
```diff
@@ -274,3 +274,15 @@
 		'sound/voice/hiss3.ogg',
 		'sound/voice/hiss4.ogg',
 	)
+
+/datum/hallucination/fake_sound/weird/radio_static
+	random_hallucination_weight = 0 // Reserved for The Freezer (& subsidiaries)
```
[the remark this answers is not in the document]
Comment (Wallemations) [reply code/modules/hallucination/fake_sound.dm]: sure thing

File: code/modules/hallucination/ice_cube.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	var/ice_duration = 6 SECONDS
+	/// Will we play the ice freeze sound?
+	var/play_ice_sound = TRUE
+
+/datum/hallucination/ice/start()
+	ice_overlay = image(ice_icon, hallucinator, ice_icon_state, ABOVE_MOB_LAYER)
+	hallucinator.client?.images |= ice_overlay
+	ADD_TRAIT(hallucinator, TRAIT_IMMOBILIZED, HALLUCINATION_TRAIT)
+	to_chat(hallucinator, span_userdanger("You're frozen in ice!"))
+	hallucinator.throw_alert(ALERT_TEMPERATURE, /atom/movable/screen/alert/cold, override = TRUE)
```
[the remark this answers is not in the document]
Comment (Wallemations) [reply code/modules/hallucination/ice_cube.dm]: I wanted to make sure it lasted the full time of the freeze, and I forgot I could pass args through the hallucination

File: code/modules/hallucination/ice_cube.dm [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/// Causes the hallucinator to believe themselves frozen in ice. Man am I glad he's frozen in there etc etc
+/datum/hallucination/ice
+	random_hallucination_weight = 3
+
+	/// What icon file to use for our hallucinator
+	var/ice_icon = 'icons/effects/freeze.dmi'
+	/// What icon state to use for our hallucinator
+	var/ice_icon_state = "ice_cube"
```
[the remark this answers is not in the document]
Comment (Wallemations) [reply code/modules/hallucination/ice_cube.dm:8]: sure thing

File: code/modules/hallucination/ice_cube.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wallemations) [reply code/modules/hallucination/ice_cube.dm:45]: oh my god I forgot to put `src.` before the vars

[the remark this answers is not in the document]
Comment (Wallemations) [reply code/modules/hallucination/ice_cube.dm]: Alright that should be all the changes in the review handled

**Comment by bruhman8ghughug** [comment]
RDS miners gonna cry

**Pull request merged**